### PR TITLE
Support OSS-Fuzz custom fields for google issue tracker.

### PR DESCRIPTION
**Pull request by oliverchang**

### CODE REVIEW

**Comment by oliverchang** [comment]
@rmistry can you please check if we're using the issue tracker correctly here with the new custom fields we're supporting here?

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py [hunk 1]
```diff
@@ -31,11 +32,15 @@
 _NUM_RETRIES = 3
 _ISSUE_TRACKER_URL = 'https://issues.chromium.org/issues'
 
+# TODO: Make these configuration settings instead of hardcoded values in code.
 # These custom fields use repeated enums.
 _CHROMIUM_OS_CUSTOM_FIELD_ID = '1223084'
```
Comment (rmistry) [inline src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py:37]: As you are likely aware but making it explicit here- If there are any assumptions about these custom fields existing in the code then they will fail in the non-Chromium tracker. Doing a quick check it should be OK, but mentioning incase you see errors about that later.

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py [hunk 2]
```diff
@@ -559,6 +570,32 @@ def _update_issue(self, new_comment=None, notify=True):
...
 
+    # Special case: OSS-Fuzz "Reported" custom field.
+    added_reported = _get_labels(self.labels.added, 'Reported-')
+    if added_reported:
+      try:
+        # Assume there is only one entry.
+        year, month, day = _parse_date_label(added_reported[0])
+        custom_field_entries.append({
+            'customFieldId': _OSS_FUZZ_REPORTED_CUSTOM_FIELD_ID,
+            'dateValue': {
```
Comment (rmistry) [inline src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py:581]: Just curious- Where did you find that this was the way to update a "Date" customfield?

Comment (rmistry) [reply src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py:581]: Looks correct to me according to this example- https://yaqs.corp.google.com/eng/q/4249097869912440832

**Comment by rmistry** [comment]
> @rmistry can you please check if we're using the issue tracker correctly here with the new custom fields we're supporting here?

LGTM (I did not notice any obvious issues)

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py (hunk 1 above)
Comment (oliverchang) [reply src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py:37]: Ack yep -- there aren't overlaps currently between Chromium and OSS-Fuzz.

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py (hunk 2 above)
Comment (oliverchang) [reply src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py:581]: Thanks! I was basing this off the service protobuf definitions, but I wasn't sure how the JSON transcoding worked.

**Pull request merged**